src/mlpack/methods/ann/layer/channel_shuffle.hpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * @file methods/ann/layer/channel_shuffle.hpp
+ * @author Abhinav Anand
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_CHANNEL_SHUFFLE_HPP
+#define MLPACK_METHODS_ANN_LAYER_CHANNEL_SHUFFLE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Definition and implementation of the Channel Shuffle Layer.
```

Comment:
I guess it makes sense to point to the ShuffleNet paper here for more information.